hpx/runtime/parcelset/parcelport_impl.hpp

```diff
@@ -811,45 +791,19 @@ namespace hpx { namespace parcelset
             sender_connection->verify(parcel_locality_id);
 #endif
             // encode the parcels
-            std::size_t num_parcels = encode_parcels(&parcels[0],
-                parcels.size(), sender_connection->buffer_,
+            encode_parcels(&p,
+                std::size_t(-1), sender_connection->buffer_,
```

Comment:
Even if you don't remove it, just passing `1` instead of the `-1` should do the right thing, no?